src/warzone/src/warzone/service/GameEngine.java

```diff
@@ -224,18 +224,19 @@ public boolean playTournament() {
 	private void prepareGameContextForTournamentMatch(String p_mapFileName) {
 		
 		d_gameContext.reset();
-		StartupService startupService = new StartupService(d_gameContext);
+		StartupService l_startupService = new StartupService(d_gameContext);
+		DominationMapReader l_dominationMapReader = new DominationMapReader(d_gameContext);
 		
-		startupService.loadMap(p_mapFileName);
+		l_dominationMapReader.loadMap(p_mapFileName);
```

Comment:
do not need to remove startupservice.loapmap() here
the better way is :  
1. keep the function  startupservice.loapmap() , mapservice.savemap(), etc
2. you have created MapReaderAdapter, DominationMapReader,ConquestMapReader, all of them are good.
2. add a property, mapType in the gamecontext, to indicate what map is used currently. it will be updated in the loadmap.
3. startupservice.loapmap() , mapservice.savemap(), editmap. just move out the current implementation into DominationMapReader. 
4. in the startupservice.loapmap()  and editmap(), check what map it is by reading the first line of the map, and then call the corresponding map reader to handle it.
5. besides that , you do not need to change the code in other class, like gameengine.java